### PR TITLE
Fix lb delete with ports

### PR DESCRIPTION
**Pull request by tylerauerbeck**
- Fixes error that occurs when attempting to delete a load balancer that has a port associated with it. Currently if you would attempt to delete an LB with an associated port you would get a FK constraint error.

### CODE REVIEW

File: internal/graphapi/loadbalancer.resolvers.go [hunk 1]
```diff
@@ -54,7 +59,47 @@ func (r *mutationResolver) LoadBalancerDelete(ctx context.Context, id gidx.Prefi
...
+
+	var (
+		err error
+		tx  *generated.Tx
+	)
+
+	tx, err = r.client.BeginTx(ctx, &sql.TxOptions{})
+	if err != nil {
+		return nil, err
+	}
```
Comment (andy-v-h) [inline internal/graphapi/loadbalancer.resolvers.go:71]:
```suggestion
	tx, err := r.client.BeginTx(ctx, &sql.TxOptions{})
	if err != nil {
		return nil, err
	}
```

File: internal/graphapi/loadbalancer.resolvers.go [hunk 2]
```diff
@@ -6,8 +6,13 @@ package graphapi
 
 import (
 	"context"
+	"database/sql"
+	"fmt"
 
+	"github.com/labstack/gommon/log"
```
Comment (andy-v-h) [inline internal/graphapi/loadbalancer.resolvers.go:12]: This is pulling in an additional logging package

File: internal/graphapi/loadbalancer.resolvers.go [hunk 3]
```diff
@@ -54,7 +59,47 @@ func (r *mutationResolver) LoadBalancerDelete(ctx context.Context, id gidx.Prefi
...
+				log.Error(fmt.Errorf("%w: %v", err, rerr).Error())
+			}
+			return nil, err
+		}
+	}
+
+	// delete loadbalancer
+	if err = tx.LoadBalancer.DeleteOneID(id).Exec(ctx); err != nil {
+		if rerr := tx.Rollback(); rerr != nil {
+			log.Error(fmt.Errorf("%w: %v", err, rerr).Error())
```
Comment (andy-v-h) [inline internal/graphapi/loadbalancer.resolvers.go:94]: This should use the resolver zap logger

File: internal/graphapi/loadbalancer.resolvers.go [hunk 4]
```diff
@@ -54,7 +59,47 @@ func (r *mutationResolver) LoadBalancerDelete(ctx context.Context, id gidx.Prefi
...
+		}
+		return nil, err
+	}
+
+	if err = tx.Commit(); err != nil {
+		if rerr := tx.Rollback(); rerr != nil {
+			log.Error(fmt.Errorf("%w: %v", err, rerr).Error())
```
Comment (andy-v-h) [inline internal/graphapi/loadbalancer.resolvers.go:101]: This should use the resolver zap logger

Comment (tylerauerbeck) [reply internal/graphapi/loadbalancer.resolvers.go]: ✅

File: internal/graphapi/loadbalancer.resolvers.go (hunk 3 above)
Comment (tylerauerbeck) [reply internal/graphapi/loadbalancer.resolvers.go]: ✅

File: internal/graphapi/loadbalancer.resolvers.go [hunk 5]
```diff
@@ -54,7 +59,47 @@ func (r *mutationResolver) LoadBalancerDelete(ctx context.Context, id gidx.Prefi
...
+		}
+		return nil, err
+	}
+
+	for _, p := range ports {
+		if err = tx.Port.DeleteOne(p).Exec(ctx); err != nil {
+			if rerr := tx.Rollback(); rerr != nil {
+				log.Error(fmt.Errorf("%w: %v", err, rerr).Error())
```
[the remark this answers is not in the document]
Comment (tylerauerbeck) [reply internal/graphapi/loadbalancer.resolvers.go]: ✅

File: internal/graphapi/loadbalancer.resolvers.go [hunk 6]
```diff
@@ -54,7 +59,47 @@ func (r *mutationResolver) LoadBalancerDelete(ctx context.Context, id gidx.Prefi
...
+	tx, err = r.client.BeginTx(ctx, &sql.TxOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// cleanup ports associated with loadbalancer
+	ports, err := tx.Port.Query().Where(predicate.Port(port.LoadBalancerIDEQ(id))).All(ctx)
+	if err != nil {
+		if rerr := tx.Rollback(); rerr != nil {
+			log.Error(fmt.Errorf("%w: %v", err, rerr).Error())
```
[the remark this answers is not in the document]
Comment (tylerauerbeck) [reply internal/graphapi/loadbalancer.resolvers.go]: ✅

File: internal/graphapi/loadbalancer.resolvers.go (hunk 2 above)
Comment (tylerauerbeck) [reply internal/graphapi/loadbalancer.resolvers.go]: ✅

File: internal/graphapi/loadbalancer.resolvers.go (hunk 1 above)
Comment (tylerauerbeck) [reply internal/graphapi/loadbalancer.resolvers.go]: ✅

**Pull request merged**